Tests/SentryTests/SentrySessionTests.m

```diff
@@ -0,0 +1,128 @@
+#import <XCTest/XCTest.h>
+#import "SentrySession.h"
+
+@interface SentrySessionTests : XCTestCase
+
+@end
+
+@implementation SentrySessionTests
+
+- (void)testInitDefaultValues {
+    SentrySession *session = [[SentrySession alloc] init];
+    XCTAssertNotNil(session.sessionId);
+    XCTAssertEqual(1, session.sequence);
+    XCTAssertEqual(0, session.errors);
+    XCTAssertTrue(session.init);
+    XCTAssertNotNil(session.started);
+    XCTAssertEqual(kSentrySessionStatusOk, session.status);
+    XCTAssertNotNil(session.distinctId);
+
+    XCTAssertNil(session.timestamp);
+    XCTAssertNil(session.releaseName);
+    XCTAssertNil(session.environment);
+    XCTAssertNil(session.duration);
+}
+
+- (void)testSerializeDefaultValues {
+    SentrySession *expected = [[SentrySession alloc] init];
+    NSDictionary<NSString *, id>  *json = [expected serialize];
+    SentrySession *actual = [[SentrySession alloc] initWithJSONObject:json];
```

Comment:
I like this. With this we test both `serialize` and `initWithJSONObject` 👏

Reply:
It does two things here which isn't ideal but it's useful to have the round trip test